Translate German comments in swhtml.cxx

leginee opened the pull request:
Okay now.
sorry for the fuzz.

please review the PR, if i missed anything.
I used deepl.com, and I hope i did not miss any lousy translations.

leginee committed:
translation of swhtml, first pass

leginee committed:
swhtml translation second pass

DamjanJovanovic commented:
It's a bit glitchy, in English it has "paragraf", "Attributs", etc.

swhtml.hxx would also be nice to translate.

Is there a way we can do these translations ourselves?

leginee commented:
> It's a bit glitchy, in English it has "paragraf", "Attributs", etc.

thanks, these glitches are my mistakes.
> 
> swhtml.hxx would also be nice to translate.

i add it too.
> 
> Is there a way we can do these translations ourselves?

deepl.com is 80+ accurate. I had to remove repetitions. Only a view times meaning is missed.
And for fun watch https://www.youtube.com/watch?v=VebSZrHmsI4

[edit:] fixed linebreaks

leginee committed:
fixed spelling paragraf, Attributs
translation of assert messages

leginee committed:
header translation first pass

leginee commented:
Translated the assert statements in body 
a rushed translation of header

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -94,14 +94,14 @@ class _HTMLAttr
-	_HTMLAttr *pPrev;	// bereits geschlossene aber noch nicht gesetze Attrs
-	_HTMLAttr **ppHead;	// der Listenkopf
+	sal_uInt16 nCount;		// Number of attrs still to be closed with a value of
+	_HTMLAttr *pNext;	// Attrs with different values still to be closed. Values
```
Maybe this is "Still to be closed attrs, with different values"

leginee replied:
This is awkward too. How about:
`open attrs, with different values`

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -204,40 +204,40 @@ struct _HTMLAttrTable
-	 AM_SPACE, 			// Abstand hart auf 0.5cm setzen
-	 AM_SOFTNOSPACE, 	// Abstand nicht setzen aber 0cm merken
-	 AM_NONE			// gar kein Append
+	 AM_NORMAL, 		// no paragraph spacing treatment
```
No changes in spacing

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -204,40 +204,40 @@ struct _HTMLAttrTable
-	 AM_SOFTNOSPACE, 	// Abstand nicht setzen aber 0cm merken
-	 AM_NONE			// gar kein Append
+	 AM_NORMAL, 		// no paragraph spacing treatment
+	 AM_NOSPACE, 		// Set distance hard to 0cm
```
Force zero spacing

leginee replied:
done

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -204,40 +204,40 @@ struct _HTMLAttrTable
-	 AM_NONE			// gar kein Append
+	 AM_NORMAL, 		// no paragraph spacing treatment
+	 AM_NOSPACE, 		// Set distance hard to 0cm
+	 AM_SPACE, 			// Set distance hard to 0.5cm
```
Force 0.5 cm spacing

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -204,40 +204,40 @@ struct _HTMLAttrTable
+	sal_Bool	bULSpaceChanged : 1;// upper/lower edge changed?
+	sal_Bool	bDfltTxtFmtColl : 1;// nTxtFmtColl is only a default
+	sal_Bool	bSpansSection : 1; 	// The context spans a SwSection
+	sal_Bool	bPopStack : 1; 		// Unstack stack elements above.
```
Delete the element above in the stack

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -382,110 +382,110 @@ class SwHTMLParser : public SfxHTMLParser, public SwClient
-	String		aStyleSource;	// Inhalt des aktuellen Style-Sheets
-	String		aContents;		// Text des akteullen Marquee, Feldes etc.
+	String		aScriptSource;	// Content of the current script block
+	String		aScriptType;	// Type of the read script (StarBasic/VB/JAVA)
```
"Type of the script"
("read" is ambiguous in English; should rather be "that we read" or "we have read" or "loaded"... but would it not be too much verbose?)

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -382,110 +382,110 @@ class SwHTMLParser : public SfxHTMLParser, public SwClient
-	_HTMLAttrContexts aContexts;// der aktuelle Attribut/Token-Kontext
-	SwHTMLFrmFmts	aMoveFlyFrms;// Fly-Frames, the anchor is moved
-	SvXub_StrLens	aMoveFlyCnts;// and the Content-Positions
+	_HTMLAttrs		aSetAttrTab;// "closed", not yet set Attr.
```
"closed" attribute, not set yet

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -382,110 +382,110 @@ class SwHTMLParser : public SfxHTMLParser, public SwClient
+	_HTMLAttrTable	aAttrTab;	// "open" attributes
+	_HTMLAttrContexts aContexts;// the current attribute/token context
+	SwHTMLFrmFmts	aMoveFlyFrms;// Fly-Frames, der Anker wird verschoben
+	SvXub_StrLens	aMoveFlyCnts;// und die Content-Positionen
```
Did you translate this _into_ German? :-)

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -382,110 +382,110 @@ class SwHTMLParser : public SfxHTMLParser, public SwClient
+	SdrObject		*pMarquee;	// current marquee
+	SwField			*pField;	// current field
+	ImageMap		*pImageMap;	// current image map
+	ImageMaps		*pImageMaps;// all read image maps
```
all image maps read so far

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -382,110 +382,110 @@ class SwHTMLParser : public SfxHTMLParser, public SwClient
-	sal_Bool bSetCrsr : 1;			// Crsr wieder auf den Anfang setzen
+	sal_Bool bDocInitialized : 1;	// Document or shell were initialized
+								// Flag by double init by recursion
+								// to prevent.
```
Flag used to avoid double initialization due to recursion

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -382,110 +382,110 @@ class SwHTMLParser : public SfxHTMLParser, public SwClient
+	sal_Bool bInFloatingFrame : 1;	// We are in a floating ::com::sun::star::frame::Frame
 	sal_Bool bInField : 1;
-	sal_Bool bKeepUnknown : 1;		// unbekannte/nicht unterstuetze Tokens beh.
+	sal_Bool bKeepUnknown : 1;		// unknown/unsupported tokens treatment
```
treatment of unknown/unsupported tokens

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -727,61 +727,61 @@ private:
 	void InsertComment( const String& rName, const sal_Char *pTag = 0 );
 
-	// sind im aktuellen Absatz ::com::sun::star::text::Bookmarks vorhanden?
+	// are there ::com::sun::star::text::bookmarks in the current paragraph?
```
are there any ....

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -828,11 +828,11 @@ private:
 
 
-	// eine neue Form beginnen
+	// start a new shape
```
start a new form

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -844,23 +844,23 @@ private:
-	// Eine Section fuer die voruebergende Aufnahme der Tabellen-Ueberschrift
-	// anlegen
+	// Create a section for the prefix of the table heading
+	// create
```
Create a section for the temporary setting of the table heading

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -895,31 +894,29 @@ private:
-	void StripTrailingPara();
 
-	// sind im aktuellen Absatz Fly-Frames vorhanden?
+	// Remove an empty paragraph at the PaM position.
```
Note you deleted a method here: void StripTrailingPara();

The following comment may be translated as:
Are there any visible Fly-Frames in the current paragraph?

ardovm commented on main/sw/source/filter/html/swhtml.hxx:
```diff
@@ -933,12 +930,12 @@ public:
                     sal_Bool bIgnoreHTMLComments = sal_False );
 
-	virtual SvParserState CallParser();   // Aufruf des Parsers
+	virtual SvParserState CallParser();   // Calling the parser
```
Call the parser

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -250,7 +250,7 @@ sal_uLong HTMLReader::Read( SwDoc &rDoc, const String& rBaseURL, SwPaM &rPam, co
 		sErr += String::CreateFromInt32((sal_Int32)xParser->GetLinePos());
 
-		// den Stream als Fehlernummer Transporter benutzen
+		// use Stream as transporter error nummber
```
use Stream to carry the error number

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -412,9 +412,9 @@ SwHTMLParser::SwHTMLParser( SwDoc* pD, const SwPaM& rCrsr, SvStream& rIn,
+						eJumpTo = JUMPTO_NONE;	// this is not valid!
 					else
-						// ansonsten ist das ein normaler (Book)Mark
+						// else it is h normal (Book)Mark
```
else it is a normal

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -440,11 +440,11 @@ __EXPORT SwHTMLParser::~SwHTMLParser()
 
-	// das DocumentDetected kann ggfs. die DocShells loeschen, darum nochmals
-	// abfragen
+	// DocumentDetected eventual deletes DocShells,
```
DocumentDetected may delete DocShells

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -582,27 +582,26 @@ SvParserState __EXPORT SwHTMLParser::CallParser()
+		// returnen at first call, show Doc
+		// and wait for Timer Callback.
+		// At this point CallParser read only one character 
+		//and a SaveState(0) called.
```
Return at first call, show Doc and wait for timer callback. At this point, CallParser read only one character and called SaveState(0)

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -621,39 +620,38 @@ void __EXPORT SwHTMLParser::Continue( int nToken )
 			NextToken( pPendStack->nToken );
 		ASSERT( !pPendStack,
-				"SwHTMLParser::Continue: Es gibt wieder einen Pend-Stack" );
+				"SwHTMLParser::Continue: a Pend-Stack exists again" );
```
there is still a Pend-Stack

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -673,15 +671,15 @@ void __EXPORT SwHTMLParser::Continue( int nToken )
 				bLFStripped = StripTrailingLF() > 0;
 
-			// noch offene Nummerierungen beenden.
+			// end open nummerations.
```
end any numberings still in progress

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -698,31 +696,31 @@ void __EXPORT SwHTMLParser::Continue( int nToken )
 		}
 
-		// den Start wieder korrigieren
+		// fix Start
```
fix ...again?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -752,12 +750,12 @@ if( pSttNdIdx->GetIndex()+1 == pPam->GetBound( sal_False ).nNode.GetIndex() )
-			// es fehlen noch ein paar Image-Map zuordungen.
-			// vielleicht sind die Image-Maps ja jetzt da?
+			// some image map assigns are missing.
+			// maybe Image-Maps do exist now?
```
some image maps were not assigned. Maybe their Image-Maps are already there?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -1055,25 +1053,25 @@ void __EXPORT SwHTMLParser::NextToken( int nToken )
-			// <SCRIPT> wird hier (von uns) ignoriert, weil es auch in
-			// Applets ignoriert wird!
+			// <SCRIPT> is ignored here (by us), because it is also ignored in
+			// applets is ignored!
```
spurious "is ignored" at the end

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -1267,8 +1263,8 @@ void __EXPORT SwHTMLParser::NextToken( int nToken )
 			InsertBodyOptions();
-			// Falls es eine Vorlage fuer die erste oder rechte Seite gibt,
-			// setzen wir die hier.
+			// If there is a template for the first or right side,
```
side -> page

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -1532,20 +1526,19 @@ void __EXPORT SwHTMLParser::NextToken( int nToken )
 		if( nOpenParaToken )
 			EndPara();
-		EndDefListItem( 0, sal_False );// <DD>/<DT> beenden und keine Vorl. setzen
+		EndDefListItem( 0, sal_False );// <DD>/<DT> quit and set no preliminaries
```
preliminaries -> template?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -2009,13 +2001,13 @@ void __EXPORT SwHTMLParser::NextToken( int nToken )
-		// Im Header muss der Inhalt von unbekannten Token ignoriert werden,
-		// es sei denn, das Token faengt mit einem '!' an.
+	    // Integrate unkown tokens to the header,
+	    // exept token starts with '!'
```
Should this rather be:
In the header we must ignore the content of unknown tokens, except for those starting with '!"
?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -2833,7 +2821,7 @@ void SwHTMLParser::_SetAttr( sal_Bool bChkEnd, sal_Bool bBeforeTable,
 		const SwFmtAnchor& rAnchor = pFrmFmt->GetAnchor();
         ASSERT( FLY_AT_PARA == rAnchor.GetAnchorId(),
-				"Nur Auto-Rahmen brauchen eine Spezialbehandlung" );
+				"Only car frames need special treatment" );
```
"car frames"? :-)
Maybe "Auto-frames"?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -2930,34 +2917,34 @@ void SwHTMLParser::NewAttr( _HTMLAttr **ppAttr, const SfxPoolItem& rItem )
+	ASSERT( ppHead, "no attribute list header found!" );
 
-	// die aktuelle Psoition als Ende-Position merken
+	// remember the current psoition as the end position
```
psoition -> position

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -2930,34 +2917,34 @@ void SwHTMLParser::NewAttr( _HTMLAttr **ppAttr, const SfxPoolItem& rItem )
-		// mehr in der Liste
+		// Then we search for the attribute that was started immediately afterwards, which
+		// has not been finished yet either (otherwise it would not be
+		// more in the list
```
more -> any more

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -2930,34 +2917,34 @@ void SwHTMLParser::NewAttr( _HTMLAttr **ppAttr, const SfxPoolItem& rItem )
-		// das Attribut nicht an der PaM-Psoition beenden, sondern da,
-		// wo das danch gestartete Attribut anfing???
+		// Do not end the attribute at the PaM psoition, but there,
+		// where the attribute started after that???
```
[...]PaM position, but there, where did the attribute start?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3053,25 +3040,24 @@ void SwHTMLParser::EndAttr( _HTMLAttr* pAttr, _HTMLAttr **ppDepAttr,
-		// gesetzt werden muessen
+		// Then do not insert, but delete. Through "fakeing" of
+		// templates by hard attributes, there may be other empty attributes in the
+		// empty attributes in the Prev list, which then must be set anyway.
```
"empty attributes" repeated twice here

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3096,43 +3082,43 @@ void SwHTMLParser::EndAttr( _HTMLAttr* pAttr, _HTMLAttr **ppDepAttr,
-		// mehr in der Liste
+		// Then we search for the attribute that was started immediately afterwards, which
+		// has not been finished yet either (otherwise it would not be
+		// more in the list
```
more -> any more

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3181,18 +3167,18 @@ void SwHTMLParser::SaveAttrTab( _HTMLAttrTable& rNewAttrTab )
-	// alle noch offenen Attribute beenden und hinter der Tabelle
-	// neu aufspannen
+	// terminate all attributes still open and span behind the table
+	// span again
```
"span again" seems to be spurious here

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3249,9 +3235,9 @@ void SwHTMLParser::SplitAttrTab( _HTMLAttrTable& rNewAttrTab,
-				// gesetzt werden muss, muessen der Previous-Attribute
-				// trotzdem gesetzt werden.
+				// If the attribute is not set before the table
+				// the Previous attribute must be set
```
"must be set" is repeated

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3261,8 +3247,8 @@ void SwHTMLParser::SplitAttrTab( _HTMLAttrTable& rNewAttrTab,
-			// den Start des Attributs neu setzen und die Verkettungen
-			// aufbrechen
+			// reset the start of the attribute and the concatenations
+			// break
```
... and break the concatenations

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3581,8 +3567,8 @@ void SwHTMLParser::NewFontAttr( int nToken )
 	Color aColor;
-	sal_uLong nFontHeight = 0;	// tatsaechlich einzustellende Font-Hoehe
-	sal_uInt16 nSize = 0;		// Fontgroesse in Netscape-Notation (1-7)
+	sal_uLong nFontHeight = 0;	// real set up font hight
```
font height to be effectively set

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3641,16 +3627,16 @@ void SwHTMLParser::NewFontAttr( int nToken )
-			// wenn die Schriftgroesse in der Ueberschrift noch
-			// nicht veraendert ist, die aus der Vorlage nehmen
+			// if the font size in the heading is not yet changed
+			// is not changed, take the one from the template.
```
extra "is not changed"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3662,8 +3648,8 @@ void SwHTMLParser::NewFontAttr( int nToken )
 
-		// in Ueberschriften wird die neue Fonthoehe wenn moeglich aus
-		// den Vorlagen geholt.
+		// in headlines, the new font height will be taken from the
```
spurious "from the"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3675,8 +3661,8 @@ void SwHTMLParser::NewFontAttr( int nToken )
-	FontFamily eFamily = FAMILY_DONTKNOW;	// Family und Pitch,
-	FontPitch ePitch = PITCH_DONTKNOW;		// falls nicht gefunden
+	FontFamily eFamily = FAMILY_DONTKNOW;	// Family and Pitch,
+	FontPitch ePitch = PITCH_DONTKNOW;		// in case not found
```
in case it is not found

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -3835,14 +3821,14 @@ void SwHTMLParser::NewPara()
-	// Styles parsen (Class nicht beruecksichtigen. Das geht nur, solange
-	// keine der CSS1-Properties der Klasse hart formatiert werden muss!!!)
+	// parse styles (ignore Class. This only works as long as
+	// none of the CSS1 properties of the class need to be hard formatted!!!)
```
hard -> forcibly

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4078,23 +4064,23 @@ void SwHTMLParser::NewTxtFmtColl( int nToken, sal_uInt16 nColl )
-		// es damit wir nicht die CLASS der PRE-Vorlage bekommen.
+		// These two tags are now mapped to the PRE template.
+		// In the case that a CLASS is specified, we delete it
+		// it so we don't get the CLASS of the PRE template.
```
"it" is repeated

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4162,18 +4148,18 @@ void SwHTMLParser::EndTxtFmtColl( int nToken )
     _HTMLAttrContext *pCntxt = PopContext( static_cast< sal_uInt16 >(nToken & ~1) );
 
-	// und noch Attribute beenden
+	// and close end Attribute
```
either "close" or "end". 
Also, remove the capital from "Attribute"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4239,22 +4225,21 @@ void SwHTMLParser::NewDefList()
-	// eines DT auf dem aktuellen Level, und die entspricht der eines
-	// DD auf dem Level davor. Fue einen Level >=2 muss also ein DD-Abstand
-	// hinzugefuegt werden
+	// The intend, which already results from a DL-, corresponds to that
```
intend -> indent

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4291,28 +4276,28 @@ void SwHTMLParser::EndDefList()
 	_HTMLAttrContext *pCntxt = PopContext( HTML_DEFLIST_ON );
 
-	// und noch Attribute beenden
+	// and close end Attribute
```
either "close" or "end"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4370,26 +4355,26 @@ void SwHTMLParser::EndDefListItem( int nToken, sal_Bool bSetColl,
 	}
 
-	// und noch Attribute beenden
+	// and close end Attribute
```
either "close" or "end".
I also think that "noch" shall be translated somehow, but I have no idea how, as I do not fully understand the context.

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4399,11 +4384,11 @@ void SwHTMLParser::EndDefListItem( int nToken, sal_Bool bSetColl,
+	//                  without text flow
+	// bSurroundOnly:   The paragraph contains at least one frame
+	//                  with a text flow but none without a text flow
+	// else:            the paragraph contains some frame
```
"some frame" -> "at least one frame"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4413,12 +4398,12 @@ sal_Bool SwHTMLParser::HasCurrentParaFlys( sal_Bool bNoSurroundOnly,
+		// - it is anchored in the current paragraph, and
+		// - every paragraph-bound frame counts, or
+		// - (only frames or text flow counts) the frame has no
+		//   text flow
```
The last two bullet points are indented to the right in the original comment. I think this has a meaning.

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4432,10 +4417,10 @@ sal_Bool SwHTMLParser::HasCurrentParaFlys( sal_Bool bNoSurroundOnly,
+				// fix #42282#: If frames with circulation are wanted,
+				// do not consider any with circulation either. These are
+				// are (still) HIDDEN controls, and it's better not to // avoid them.
+				// better not to avoid them.
```
Watch out for repetitions here

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4479,16 +4464,16 @@ const SwFmtColl *SwHTMLParser::GetCurrFmtColl() const
-	sal_uInt16 nLeftMargin = 0, nRightMargin = 0;	// die Einzuege und
-	short nFirstLineIndent = 0;					// Abstaende
+	sal_uInt16 nLeftMargin = 0, nRightMargin = 0;	    // the margins and
+	short nFirstLineIndent = 0;					        // Intend
```
Intend -> Indent

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4528,14 +4513,13 @@ void SwHTMLParser::SetTxtCollAttrs( _HTMLAttrContext *pContext )
-
+				// if now another template should be set than
+				// previously, the previous template must be replaced by hard attributes
+				// must be replaced
```
Watch for repetition of "must be replaced"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4528,14 +4513,13 @@ void SwHTMLParser::SetTxtCollAttrs( _HTMLAttrContext *pContext )
-					// die Attribute, die die bisherige Vorlage setzt
-					// hart einfuegen
+					// the attributes set by the previous template
+					// insert hard
```
Force the attributes set by the previous template

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4546,17 +4530,16 @@ void SwHTMLParser::SetTxtCollAttrs( _HTMLAttrContext *pContext )
 			else
 			{
-				// hart Attributieren
+				// set attributes hard
```
Force attributes

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4730,15 +4713,15 @@ void SwHTMLParser::NewCharFmt( int nToken )
     _HTMLAttrContext *pCntxt = new _HTMLAttrContext( static_cast< sal_uInt16 >(nToken) );
 
-	// die Vorlage setzen und im Kontext merken
+	// Set the template und im Kontext merken
```
Set the template and store it in the context

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4819,18 +4802,18 @@ void SwHTMLParser::InsertSpacer()
 			SetAnchorAndAdjustment( eVertOri, eHoriOri, aFrmSet );
 
-			// und noch die Groesse des Rahmens
+			// and still the size of the frame
```
"still" -> "again" ? or "also"?

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4865,21 +4848,21 @@ void SwHTMLParser::InsertSpacer()
+				// change the lower paragraph spacing of the previous node,
+				// if possible
 
 				SetAttr();	// noch offene Absatz-Attribute setzen
```
Set more open paragraph attributes

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -4865,21 +4848,21 @@ void SwHTMLParser::InsertSpacer()
-				// Abstand erzeugt.
+				// If the paragraph in front of it is not a txttenode, then now
+				// an empty paragraph is created which already creates a line height
+				// spacing is created.
```
Remove the last "is created"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -5004,27 +4987,27 @@ void SwHTMLParser::InsertIDOption()
+	//     text flow
+	// 4.) otherwise a left aligned frame receives a right
+	//     "anchor only" text flow and right aligned frame receives a
+	//      left anchor only" text flow.
```
Quotes missing between "left" and "anchor"

ardovm commented on main/sw/source/filter/html/swhtml.cxx:
```diff
@@ -5136,18 +5119,18 @@ void SwHTMLParser::InsertLineBreak()
+		// MIB 21.02.97: Actually, the lower paragraph should be set to 0.
+		// But this works with something like <BR ..><P>.
+		// (>Netacpe). Therefore we leave it for now.
 		AppendTxtNode( AM_NOSPACE );
```
If a **Clear** was executed in a non-empty paragraph, **then** a new paragraph must be opened afterwards.
MIB 21.02.97: actually, the lower paragraph **spacing** should be set to 0. But this works with something like `<BR ..><P>` (>**Netscape**). Therefore we leave it for now.

ardovm reviewed (changes requested):
Thank you very much for your hard work!
I added some proposed corrections, based on my understanding of the original comments.
IMHO this PR is not ready for merging because of a problem in the header file `swhtml.hxx` around line 900.

leginee committed:
Update swhtml.hxx 

- fixed code spacing issues
- Force zero spacing
- changed translation 
       Number of attrs still to be closed with a value of 
  to
      open attrs, with different values

leginee committed:
fixed various translations

leginee committed:
Update various translations

leginee committed:
Update various translations

leginee commented:
Thanks Arrigo for the review. Very nice suggestions.

@DamjanJovanovic , @ardovm 
can you check [these lines please](https://github.com/apache/openoffice/blob/cd8df0c879df4c07f6fa5a97c3d5f3ceaa9d50b5/main/sw/source/filter/html/swhtml.hxx#L797-L807)

the code looks odd. I think there is an ");" to much in line 802
but this is not a translation issue.

ardovm commented:
> @DamjanJovanovic , @ardovm can you check [these lines please](https://github.com/apache/openoffice/blob/cd8df0c879df4c07f6fa5a97c3d5f3ceaa9d50b5/main/sw/source/filter/html/swhtml.hxx#L797-L807)
> 
> the code looks odd. I think there is an ");" to much in line 802 but this is not a translation issue.

That ");" seems legitimate to me: it marks the end of the declaration of method `InsertDrawObject`
The declaration of the next method `InsertControl` starts at the next line.

leginee commented:
Ahhh now I get it. the intend got me all mixed up.

leginee committed:
fixed indent of function InsertControl

leginee commented:
okay fixed that too. I hope this is now complete?

ardovm commented:
To be honest, I still have open comments on the .cxx file... did you look at them?

leginee committed:
Update translation swhtml.cxx

leginee committed:
Update translations swhtml.hxx

leginee committed:
Update translation swhtml.cxx

leginee commented:
okay, I think I have now looked into all, but I am unsure. Some I left open and only commented them.

ardovm reviewed (approved):
Looks good to me!

I suggest we "squash and merge".

leginee commented:
done. Thanks to all participating
